specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/config/cache.options.extract.headers.yaml

```
@@ -0,0 +1,67 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+---
+name: test
+catalogs:
+  test0:
+    type: test
+    options:
+      id: 1
+      schema: |
+        {
+          "type": "object",
+          "properties": {
+            "correlationId": {
+              "type": "integer"
+            },
+            "status": {
+              "type": "string"
+            }
+          },
+          "required": [
+            "correlationId",
+            "status"
+          ]
+        }
+bindings:
+  app0:
+    type: kafka
+    kind: cache_client
+    routes:
+      - exit: cache0
+        when:
+          - topic: test
+  cache0:
+    type: kafka
+    kind: cache_server
+    options:
+      topics:
+        - name: test
+          headers:
+            - correlation-id: $.correlationId
```

I realized this syntax for headers is inconsistent with the other fields in the same object, like `key` and `value`.

In those case, each one provides a model, constraining the type, so if we would want to add that capability later for headers we would naturally have an object for headers, with header names as keys, and `model` in the value.

Then we would also be able to add `extract` or `value` as another property, indicating an expression, with support for short form in the meantime.

Also, even though we know we are using JSON path internally, it is inconsistent with our other expressions in `zilla.yaml`, so perhaps we want to use the following instead?

```suggestion
          headers:
            correlation-id: ${value.correlationId}
```
and internally treat this as `$.correlationId` but without exposing it directly in the `zilla.yaml` syntax.